packages/textfield/src/component.tsx

```diff
@@ -28,8 +27,10 @@ export const TextField = forwardRef<HTMLInputElement, TextFieldProps>(
     } = props;
 
     activateI18n(enMessages, nbMessages, fiMessages);
-
-    const id = useId(providedId);
+    
+    // useId returns a string that includes colons (:), e.g., :r0:, :r1:, etc.
+    // This string is NOT supported in CSS selectors. Hence the replace.
+    const id = providedId ?? useId().replace(/:/g, '');
```

Comment:
Not sure why the original `useId` util had to rely on `useLayoutEffect`, but would it make sense to move the React `useId()` hook to `src/useId.ts` and simplify the previous `useId` util? I'm mostly thinking about reusing `generateId` as prefix, which would ensure unique ids in the case where we have multiple React apps using multiple TextField components on one page. I found an article about that here: https://hetdesai03.medium.com/a-complete-guide-to-useid-hook-in-react-18-22119ecfd87f Wdyt?

Reply:
I think that old `useId` solution exists from react 16 days. This hook was added in React 18. 

Notice also our `useId` hook used `useLayoutEffect` which is a performance pitfall, but it was common to use during React 16 days.

I'd say, we should replace instances of our `useId` with React's `useId` but maybe not in this PR.

Reply:
I see. But what about `generateId` prefix util that would increase uniqueness of the id? If React's `useId` only generates `r0`, `r1` strings, the chance of multiple TextFields across multiple React apps having the same id is higher. That's why I was thinking that we could maybe prefix those ids with whatever `generateId` returns, the same way it was done in the utils.

Reply:
Hmm, the docs says that react will always generate unique id's for each useId within the vdom. Do you think there might be a situation where this won't hold? 

> multiple TextFields across multiple React apps
Do you mean in case of podlets?

Reply:
> Do you mean in case of podlets?
Yes, I was thinking about that :) If there were multiple React apps on a layout/page, served as podlets, then we could theoretically have multiple text fields with the same id, e.g. `r0`.

Reply:
Okay, so I have updated our useId hook and tested it with other components :)